subprojects/core/src/main/java/org/gradle/internal/operations/DefaultBuildOperationExecutor.java

```diff
@@ -63,7 +65,8 @@ public DefaultBuildOperationExecutor(
             currentBuildOperationRef
         );
         this.buildOperationQueueFactory = buildOperationQueueFactory;
-        this.fixedSizePool = executorFactory.create("Build operations", parallelismConfiguration.getMaxWorkerCount());
+        managedExecutors.put(BuildOperationConstraint.MAX_WORKERS, executorFactory.create("Build operations", parallelismConfiguration.getMaxWorkerCount()));
+        managedExecutors.put(BuildOperationConstraint.UNCONSTRAINED, executorFactory.create("Unconstrained build operations"));
```

Comment:
I wonder about still having a constraint there, although a much higher one.
Is the effect still beneficial with something like maxWorkerCount * 10?

I understand that for HTTP requests, the underlying code handles it. But this is general operation execution and that statement may not hold always.

Reply:
Yeah, I thought about that - artifact transforms are completely unbounded IIRC and because this is a cached pool and threads will be reused thought it'd likely be fine. maxWorkerCount * 10 is probably not unreasonable though - I'll make that change.